### PR TITLE
Implement a new item drops system

### PR DESCRIPTION
**Pull request by eternalcomet**
## Description
Implment a new drop system,including chest rewards, boss chest rewards, monster drop, investigation points, BGM CDs collection,etc.
I renamed old DropSystem to DropSystemLegacy, if new drop system failed to find necessary resources it will fallback to the legacy one.

Please carefully read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md) before making any pull requests.

## Type of changes

- [ ] Bug fix
- [x] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by eternalcomet** [comment]
Maybe need to clean lombok annotations in GameData.java, @KingRainbow44 you can do it since i'm not familiar with that.

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]
please replace `onMethod = @__(@Override)` with the JDK 8 version which uses `onMethod_ = @Override`

**Comment by KingRainbow44** [comment]
additionally, run `./gradlew spotlessFormat` to format your code!

**Comment by eternalcomet** [comment]
I'll do it later.

File: src/main/java/emu/grasscutter/command/commands/ReloadCommand.java [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
+public final class ReloadCommand implements CommandHandler {
+
+    @Override
+    public void execute(Player sender, Player targetPlayer, List<String> args) {
+        CommandHandler.sendMessage(sender, translate(sender, "commands.reload.reload_start"));
+
+        Grasscutter.loadConfig();
+        Grasscutter.loadLanguage();
+        Grasscutter.getGameServer().getGachaSystem().load();
+        Grasscutter.getGameServer().getDropSystemLegacy().load();
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/command/commands/ReloadCommand.java]: Please reload the new drop system, or remove the existing drop system from the reload command.

File: src/main/java/emu/grasscutter/data/GameData.java [hunk 2]
```diff
@@ -197,7 +197,8 @@ public final class GameData {
     @Getter
     private static final Int2ObjectMap<DailyDungeonData> dailyDungeonDataMap =
             new Int2ObjectOpenHashMap<>();
-
+    @Getter private static final Int2ObjectMap<DropTableData> dropTableDataMap=new Int2ObjectOpenHashMap<>();
+    @Getter private static final Int2ObjectMap<DropMaterialData> dropMaterialDataMap=new Int2ObjectOpenHashMap<>();
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/data/GameData.java]:
```suggestion
    @Getter private static final Int2ObjectMap<DropTableData> dropTableDataMa p= new Int2ObjectOpenHashMap<>();
    @Getter private static final Int2ObjectMap<DropMaterialData> dropMaterialDataMap = new Int2ObjectOpenHashMap<>();
```

File: src/main/java/emu/grasscutter/data/excels/DropMaterialData.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package emu.grasscutter.data.excels;
+
+import emu.grasscutter.data.GameResource;
+import emu.grasscutter.data.ResourceType;
+import lombok.Getter;
+
+@ResourceType(name = "DropMaterialExcelConfigData.json")
+@Getter
+public class DropMaterialData extends GameResource {
+    @Getter(onMethod = @__(@Override))
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/data/excels/DropMaterialData.java]:
```suggestion
    @Getter(onMethod_ = @Override)
```

File: src/main/java/emu/grasscutter/data/excels/DropTableData.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import emu.grasscutter.data.ResourceType.LoadPriority;
+import emu.grasscutter.data.common.DropItemData;
+import lombok.Getter;
+
+import java.util.List;
+
+@ResourceType(name ={"DropTableExcelConfigData.json","DropSubTableExcelConfigData.json"} , loadPriority = LoadPriority.HIGH)
+@Getter
+public class DropTableData extends GameResource {
+    @Getter(onMethod = @__(@Override))
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/data/excels/DropTableData.java]:
```suggestion
    @Getter(onMethod_ = @Override)
```

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]

File: src/main/java/emu/grasscutter/scripts/data/SceneMonster.java [hunk 5]
```diff
@@ -5,11 +5,14 @@
 
 @ToString
 @Setter
-public class SceneMonster extends SceneObject {
+public class SceneMonster extends SceneObject{
+	public int config_id;
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/scripts/data/SceneMonster.java]:
```suggestion
    public int config_id;
```

File: src/main/java/emu/grasscutter/scripts/data/SceneMonster.java [hunk 6]
```diff
@@ -5,11 +5,14 @@
...
     public int monster_id;
-    public int pose_id;
-    public int drop_id;
-    public boolean disableWander;
+	public int pose_id;
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/scripts/data/SceneMonster.java]:
```suggestion
    public int pose_id;
```

File: src/main/java/emu/grasscutter/scripts/data/SceneMonster.java [hunk 7]
```diff
@@ -5,11 +5,14 @@
...
+    public int level;
+	public int drop_id;
+	public boolean disableWander;
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/scripts/data/SceneMonster.java]:
```suggestion
    public int drop_id;
    public boolean disableWander;
```

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]
Please run `./gradlew spotlessApply`

**Comment by eternalcomet** [comment]
> Please run `./gradlew spotlessApply`

done

**Comment by eternalcomet** [comment]
I have a issue with .gitattribute file. when i checkout, i will see a tons of changes for crlf or lf, no matter what core.autocrlf is set. so maybe you can run spotlessApply yourself. sorry for that

**Review by KingRainbow44 — APPROVED** [review]

**Pull request merged**